File: admin-client/src/services/development_tool/MonitorEFCoreService.ts
import Http from "@/core/utils/Http";

/**
 * EFCore 监控服务
 */
export default class MonitorEFCoreService {
  static urlPrefix = "/api/v1/admin/MonitorEFCore";

  /**
   * 获取 EFCore 监控上下文
   *
   */
  static getEFCoreMonitorContext() {
    return Http.post(`${this.urlPrefix}/GetEFCoreMonitorContext/`);
  }

  /**
   * 获取 EFCore Sql 监控上下文 耗时排行榜
   *
   */
  static getTimeConsumingRanking() {
    return Http.post(`${this.urlPrefix}/GetTimeConsumingRanking/`);
  }

  /**
   * 获取 EFCore Sql 监控上下文 最新数据
   *
   */
  static getNewest() {
    return Http.post(`${this.urlPrefix}/GetNewest/`);
  }
}

File: admin-server/HZY.Api.Admin/ApplicationServices/DevelopmentTools/MonitorEFCore/Impl/MonitorEFCoreService.cs
namespace HZY.Api.Admin.ApplicationServices.DevelopmentTools.MonitorEFCore.Impl;

/// <summary>
/// 监控 EFCore 服务
/// </summary>
public class MonitorEFCoreService : IMonitorEFCoreService
{
    /// <summary>
    /// GetEFCoreMonitorContext
    /// </summary>
    /// <returns></returns>
    public EntityFrameworkRepositoriesMonitorContext? GetEFCoreMonitorContext()
    {
        return RepositoryEntityFrameworkMonitorCache.Context;
    }

    /// <summary>
    /// GetNewest
    /// </summary>
    /// <returns></returns>
    public List<EntityFrameworkRepositoriesMonitorSqlContext>? GetNewest()
    {
        return RepositoryEntityFrameworkMonitorCache.SqlContext?.OrderByDescending(w => w.Time).Take(20).ToList();
    }

    /// <summary>
    /// GetTimeConsumingRanking
    /// </summary>
    /// <returns></returns>
    public List<EntityFrameworkRepositoriesMonitorSqlContext>? GetTimeConsumingRanking()
    {
        return RepositoryEntityFrameworkMonitorCache.SqlContext?.OrderByDescending(w => w.ElapsedMilliseconds).Take(20).ToList();
    }



}
